kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java

```diff
@@ -113,6 +115,25 @@ public TransactionCoordinator getTransactionCoordinator(String tenant) {
         return transactionCoordinatorByTenant.computeIfAbsent(tenant, this::createAndBootTransactionCoordinator);
     }
 
+    @Override
+    public ReplicaManager getReplicaManager(String tenant) {
+        return replicaManagerByTenant.computeIfAbsent(tenant, s -> {
+            Optional<TransactionCoordinator> transactionCoordinatorOptional = Optional.empty();
+            if (kafkaConfig.isEnableTransactionCoordinator()) {
+                transactionCoordinatorOptional = Optional.of(getTransactionCoordinator(tenant));
+            }
+            try {
+                return new ReplicaManager(kafkaConfig,
+                        Time.SYSTEM,
+                        transactionCoordinatorOptional,
+                        producePurgatory);
+            } catch (Exception e) {
+                log.error("Failed to init ReplicaManager for tenant {}", tenant, e);
+                throw new IllegalStateException(e);
+            }
```

Comment:
Remove the try-catch block. The exception could only be thrown from `EntryFormatter#create`. If you want to fail fast, you can create the `EntryFormatter` in advance.

Reply:
Good idea.